From f99a0a4d081f8086ab00c4fda33d116d0a0747c1 Mon Sep 17 00:00:00 2001
From: Vishesh Handa <me@vhanda.in>
Date: Tue, 1 Apr 2014 16:38:37 +0200
Subject: [PATCH 31/48] FileIndexingJob: Remember to disconnect the process
 before deleting it

Otherwise if this is called when the process is still running it results
in the slotIndexedFile function being called when the process is finally
destroyed.

This fixes the testTimeout auto test
---
 src/file/fileindexingjob.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/file/fileindexingjob.cpp b/src/file/fileindexingjob.cpp
index a6f8183..1d5f6f7 100644
--- a/src/file/fileindexingjob.cpp
+++ b/src/file/fileindexingjob.cpp
@@ -96,6 +96,7 @@ void FileIndexingJob::slotIndexedFile(int, QProcess::ExitStatus exitStatus)
 {
     // stop the timer since there is no need to kill the process anymore
     m_processTimer->stop();
+    m_process->disconnect(this);
     m_process->deleteLater();
     m_process = 0;
 
@@ -141,6 +142,7 @@ void FileIndexingJob::slotIndexedFile(int, QProcess::ExitStatus exitStatus)
 void FileIndexingJob::slotProcessTimerTimeout()
 {
     // Emulate a crash so that we narrow down the file which is taking too long
+    qDebug() << "Process took too long killing";
     slotIndexedFile(1, QProcess::CrashExit);
 }
 
-- 
1.9.1

